pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -531,19 +560,74 @@ func (p *provider) Get(machine *v1alpha1.Machine) (instance.Instance, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to get datacenter finder: %v", err)
 	}
-	virtualMachine, err := finder.VirtualMachine(context.TODO(), machine.Spec.Name)
+	virtualMachine, err := finder.VirtualMachine(ctx, machine.Spec.Name)
 	if err != nil {
 		if err.Error() == fmt.Sprintf("vm '%s' not found", machine.Spec.Name) {
 			return nil, cloudprovidererrors.ErrInstanceNotFound
 		}
 		return nil, fmt.Errorf("failed to get server: %v", err)
 	}
 
-	powerState, err := virtualMachine.PowerState(context.TODO())
+	powerState, err := virtualMachine.PowerState(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get powerstate: %v", err)
 	}
 
+	// Check if the creationCompleteFieldName is set to machine.Spec.Name.
+	// If that is not the case, the creation didn't complete successfully and
+	// we must delete the instance so it gets recreated
+	fieldValue, err := getValueForField(ctx, virtualMachine, creationCompleteFieldName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get value for field: %v", err)
+	}
+	if fieldValue != machine.Spec.Name {
+		// TODO: This should leverage .Cleanup, but we can currently not do that,
+		// because .Cleanup needs cloud.MachineCreateDeleteData which we don't have here
```

Comment:
We should at least extract the deletion logic interaction into a dedicated function so we reuse it

Reply:
we need to make the `detach PVs` part configurable then, because we dont have a client to get PVs in the `Get`